src/Course/Extend.hs

```diff
@@ -51,8 +50,9 @@ instance Extend List where
     (List a -> b)
     -> List a
     -> List b
-  (<<=) =
-    error "todo: Course.Extend (<<=)#instance List"
+  _    <<= Nil           = Nil
+  func <<= l@(_ :. rest) = func l :. (func <<= rest)
+  -- REVIEW: is this the "right way" to do this, or should I be using foldr or something?
```

Comment:
I think recursion is perfectly fine in a vacuum.
However, in some cases, it makes the code look needlessly complex, and in these cases you'd use fold - i.e. `foldl (+) xs` is way more concise than a recursive analog.
But this is not one of those cases.

Reply:
When approaching this problem, would you have reached for recursion, or some other looping function?

Reply:
I would have gone with recursion